1_imagestream_data_gen/imagestream_data_gen.py

```diff
@@ -0,0 +1,140 @@
+import json
+import os
+
+import urllib3
+import yaml
+from kubernetes import config
+from openshift.dynamic import DynamicClient
+
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+script_dir = os.path.dirname(os.path.realpath(__file__))
+output_dir = os.path.join(script_dir, '../output')
+
+try:
+    k8s_client = config.new_client_from_config()
+    dyn_client = DynamicClient(k8s_client)
+except Exception as e:
+    print("\n[!] Failed while setting up OpenShift client. Ensure KUBECONFIG is set. ")
+    print(e)
+    exit(1)
+
+# Ensure KUBECONFIG is set to source cluster by checking for OpenShift v3 in version endpoint
+try:
+    kube_minor_version = dyn_client.version.get("kubernetes", {}).get("minor", "").split("+")[0]
+    if int(kube_minor_version) > 11:
+        print(
+            "\n[!] [WARNING] KUBECONFIG should be set to source cluster (likely OCP 3.x) for 'Stage 1', but OCP 4.x "
+            "cluster detected.")
```

Comment:
@sseago I believe you can still run a 4.2->4.5 migration just ignoring the warning. A lot of us were burnt on making sure we have the right kube-context, so if you dont mind, I would want this warning to be present

Reply:
@alaypatel07 If the goal is to avoid messing up and getting the wrong kubecontext, the warning is better than nothing, but it won't help with getting the wrong context in the 4.x->4.x case, since you'll see the same warning whether or not it's there. That being said, I'm not sure how else to determine whether the context is right unless you could explicitly list src and dest kubeconfigs in a shared configuration file and pull from there.

Reply:
@sseago may be in the future we could do something like take a migplan as input and generate kubeconfigs and namespaces to migrate from there avoiding any mess-ups! But I would defer it until someone complains about 4.x->4.x